addons/website/static/src/snippets/s_searchbar/000.js

```diff
@@ -240,10 +243,15 @@ publicWidget.registry.searchBar = publicWidget.Widget.extend({
                 break;
             case $.ui.keyCode.UP:
             case $.ui.keyCode.DOWN:
-                ev.preventDefault();
                 if (this.$menu) {
-                    let $element = ev.which === $.ui.keyCode.UP ? this.$menu.children().last() : this.$menu.children().first();
-                    $element.focus();
+                    const focusedEl = this.$menu && this.$menu[0].querySelector('.dropdown-item:focus');
+                    const suggestionEls = [null, ...this.$menu[0].querySelectorAll('.dropdown-item')];
```

Comment:
- You do `this.$menu &&` inside a `if (this.$menu) {`
- Nitpicking: Instead of searching twice for dropdown-items, can't you invert the order?
```javascript
const suggestionEls = [null, ...this.$menu[0].querySelectorAll('.dropdown-item')];
const focusedEl = suggestionEls.find(el => el.matches(':focus'));
```
Maybe even better:
- Not searching for dropdown-item but just taking all children of the menu
- Not searching for the focused element
```javascript
const suggestionEls = [null, ...this.$menu[0].children];
const focusedEl = document.activeElement;
```
?